pkg/phlaredb/block_querier.go

```diff
@@ -561,28 +559,29 @@ func (q Queriers) MergeProfilesStacktraces(ctx context.Context, stream *connect.
 			if err != nil {
 				return err
 			}
-			lock.Lock()
-			defer lock.Unlock()
-			result = append(result, merge)
+			m.MergeStackTraces(merge.Stacktraces, merge.FunctionNames)
```

Comment:
Wouldn't it be better if we return the tree directly from the `q.MergeByStacktraces` ?

Reply:
This is a good point. I'd experiment with this. My idea was to merge samples to the tree (in batches) directly in `MergeByStacktraces`, and pass the merger as an argument. However, I wasn't not 100% sure if this approach is more efficient, and I didn't want to delay shipping of this feature